crates/bevy_core/src/time/time.rs

```diff
@@ -1,95 +1,264 @@
     #[inline]
     pub fn delta_seconds(&self) -> f32 {
         self.delta_seconds
     }
 
-    /// The delta between the current and last tick as [`f64`] seconds
+    /// Returns how much time has advanced since the last update, as [`f64`] seconds.
     #[inline]
     pub fn delta_seconds_f64(&self) -> f64 {
         self.delta_seconds_f64
     }
 
-    /// The time from startup to the last update in seconds
+    /// Returns the exact CPU time elapsed since the last update, as a [`Duration`].
     #[inline]
-    pub fn seconds_since_startup(&self) -> f64 {
+    pub fn raw_delta(&self) -> Duration {
+        self.raw_delta
+    }
+
+    /// Returns the exact CPU time elapsed since the last update, as [`f32`] seconds.
+    #[inline]
+    pub fn raw_delta_seconds(&self) -> f32 {
+        self.raw_delta_seconds
+    }
+
+    /// Returns the exact CPU time elapsed since the last update, as [`f64`] seconds.
+    #[inline]
+    pub fn raw_delta_seconds_f64(&self) -> f64 {
+        self.raw_delta_seconds_f64
+    }
+
+    /// Returns how much time has advanced since startup, as [`Duration`].
+    #[inline]
+    pub fn elapsed_since_startup(&self) -> Duration {
+        self.elapsed_since_startup
+    }
+
+    /// Returns how much time has advanced since startup, as [`f32`] seconds.
+    #[inline]
+    pub fn seconds_since_startup(&self) -> f32 {
         self.seconds_since_startup
     }
 
-    /// The [`Instant`] the app was started
+    /// Returns how much time has advanced since startup, as [`f64`] seconds.
     #[inline]
-    pub fn startup(&self) -> Instant {
-        self.startup
+    pub fn seconds_since_startup_f64(&self) -> f64 {
+        self.seconds_since_startup_f64
     }
 
-    /// The ['Instant'] when [`Time::update`] was last called, if it exists
+    /// Returns the exact CPU time elapsed since startup, as [`Duration`].
     #[inline]
-    pub fn last_update(&self) -> Option<Instant> {
-        self.last_update
+    pub fn raw_elapsed_since_startup(&self) -> Duration {
+        self.raw_elapsed_since_startup
+    }
+
+    /// Returns the exact CPU time elapsed since startup, as [`f32`] seconds.
+    #[inline]
+    pub fn raw_seconds_since_startup(&self) -> f32 {
+        self.raw_seconds_since_startup
     }
 
-    /// The ['Duration'] from startup to the last update
+    /// Returns the exact CPU time elapsed since startup, as [`f64`] seconds.
     #[inline]
-    pub fn time_since_startup(&self) -> Duration {
-        self.time_since_startup
+    pub fn raw_seconds_since_startup_f64(&self) -> f64 {
+        self.raw_seconds_since_startup_f64
     }
 }
 
-pub(crate) fn time_system(mut time: ResMut<Time>) {
-    time.update();
+pub(crate) fn update_time(
+    mut time: ResMut<Time>,
+    fixed_time: Option<Res<FixedTime>>,
+    accumulator: Option<ResMut<FixedTimestepState>>,
+) {
+    let first = time.last_update().is_none();
+    let now = Instant::now();
+    time.update_with_instant(now);
```

Comment:
you could use `update()` directly here

Reply:
I used `update_with_instant` so that `Time` and `FixedTime` stay synced. Consecutive calls to `Instant::now()` always gives a slightly different value.

It does make the function look weird, even though we only check on the first frame.

Reply:
Oh, right, I missed the part where you were reusing the now variable later